src/main/scala/org/apache/spark/sql/hbase/HBaseSQLParser.scala

```diff
@@ -74,6 +80,34 @@ class HBaseSQLParser extends SqlParser {
         InsertValueIntoTableCommand(tableName, valueStringSeq)
     }
 
+  // Standard Syntax:
+  // UPDATE tablename SET column = value [, column = value ...] [WHERE expression]
+  protected lazy val update: Parser[LogicalPlan] =
+    (UPDATE ~> relation <~ SET) ~ rep1sep(updateColumn, ",") ~ (WHERE ~> expression) ^^ {
+      case table ~ updateColumns ~ exp =>
+        val (columns, values) = updateColumns.unzip
+        catalyst.logical.UpdateTable(
+          table.asInstanceOf[UnresolvedRelation].tableName,
+          columns.map(UnresolvedAttribute.quoted),
+          values,
+          Filter(exp, table))
+    }
+
+  protected lazy val updateColumn: Parser[(String, String)] =
+    ident ~ (EQ ~> ident) ^^ {
+      case column ~ value => (column, value)
+    }
+
+  // Standard Syntax:
+  // DELETE FROM tablename [WHERE expression]
+  protected lazy val delete: Parser[LogicalPlan] =
+    DELETE ~ FROM ~> relation ~ (WHERE ~> expression) ^^ {
+      case table ~ exp =>
+        catalyst.logical.DeleteFromTable(
+          table.asInstanceOf[UnresolvedRelation].tableName,
+          Filter(exp, table))
```

Comment:
@scwf  The projection should include the references from the "exp". I'm not sure if all columns are included by default. Even if so, this is a waste. HBase tables may be very wide and selecting unnecessary columns could mean a big performance hit. We should have projections that are just references in the "exp" unioned with all key columns.